Gordon360/Services/ActivityService.cs

```diff
@@ -308,6 +312,7 @@ public async Task<ACT_INFO> UpdateActivityImageAsync(ACT_INFO involvement, IForm
             // Put current DateTime in filename so the browser knows it's a new file and refreshes cache
             var filename = $"canvasImage_{DateTime.Now:yyyy-MM-dd_HH-mm-ss-fff}.png";
             var imagePath = Path.Combine(_webHostEnvironment.ContentRootPath, "browseable", "uploads", involvement.ACT_CDE.Trim(), filename);
+            var url = $"{ _server.Features.Get<IServerAddressesFeature>()?.Addresses.FirstOrDefault(a => a.Contains("https"))}/browseable/uploads/{involvement.ACT_CDE.Trim()}/{filename}";
```

Comment:
I would recommend splitting this out for code quality and readability.
```suggestion
            var baseUrl = _server.Features.Get<IServerAddressesFeature>()?.Addresses.FirstOrDefault(a => a.Contains("https"));
            var url = $"{baseUrl}/browseable/uploads/{involvement.ACT_CDE.Trim()}/{filename}";
```